### DISCUSSION TITLE
how to acess a module's api?

### DISCUSSION THREAD

**Discussion by dark-ether** [nvim-neorg/neorg#343] [Q&A] [answered]
for instance if i wanted to use the core.mode get_ mode() function how do  i do that? i tried require
("neorg").public.core.mode.get_mode(), require("neorg").core.mode.get_mode() , require("neorg").public.get_mode()
, require("neorg.core.mode"), require("neorg/core/mode"), require("neorg/modules/mode"), neither of these worked

**Comment by vhyrro** [comment, marked as answer]
Hey! To do that you simply do `require("neorg").modules.get_module("core.mode").get_mode()` or alternatively `require("neorg.modules").get_module("core.mode").get_mode()`.

You can check out all the functions for accessing different parts of the module system in the `lua/neorg/modules.lua` file, almost all of them are documented :P, hope that helps!
